api/scripts/wazuh-apid.py

```diff
@@ -59,49 +26,90 @@ def start(foreground, root, config_file):
     config_file : str
         Path to the API config file
     """
-    configuration.api_conf.update(configuration.read_yaml_config(config_file=args.config_file))
+    import asyncio
+    import logging
+    import os
+    import ssl
+
+    import connexion
+    import uvloop
+    from aiohttp_cache import setup_cache
+
+    import wazuh.security
+    from api import __path__ as api_path
+    # noinspection PyUnresolvedReferences
+    from api import validator
+    from api.api_exception import APIError
+    from api.constants import CONFIG_FILE_PATH
+    from api.middlewares import set_user_name, security_middleware, response_postprocessing
+    from api.uri_parser import APIUriParser
+    from api.util import to_relative_path
+    from wazuh.core import pyDaemonModule
+
+    configuration.api_conf.update(configuration.read_yaml_config(config_file=config_file))
     api_conf = configuration.api_conf
-    cors = api_conf['cors']
     log_path = api_conf['logs']['path']
 
+    # Set up logger
+    set_logging(log_path=log_path, debug_mode=api_conf['logs']['level'], foreground_mode=foreground)
+    logger = logging.getLogger('wazuh')
+
+    # Set correct permissions on api.log file
+    if os.path.exists(os.path.join(common.ossec_path, log_path)):
+        os.chown(os.path.join(common.ossec_path, log_path), common.ossec_uid(), common.ossec_gid())
+        os.chmod(os.path.join(common.ossec_path, log_path), 0o660)
+
+    # Configure https
     ssl_context = None
-    if api_conf['https']['enabled'] and os.path.exists(api_conf['https']['key']) and \
-            os.path.exists(api_conf['https']['cert']):
+    if api_conf['https']['enabled']:
+        # Generate SSL if it does not exist and HTTPS is enabled
+        if not os.path.exists(api_conf['https']['key']) or not os.path.exists(api_conf['https']['cert']):
+            logger.info('HTTPS is enabled but cannot find the private key and/or certificate. '
+                        'Attempting to generate them.')
+            private_key = configuration.generate_private_key(api_conf['https']['key'])
+            logger.info(f"Generated private key file in WAZUH_PATH/{to_relative_path(api_conf['https']['key'])}.")
+            configuration.generate_self_signed_certificate(private_key, api_conf['https']['cert'])
+            logger.info(f"Generated certificate file in WAZUH_PATH/{to_relative_path(api_conf['https']['cert'])}.")
```

Comment:
```suggestion
        if not os.path.exists(api_conf['https']['key']) or not os.path.exists(api_conf['https']['cert']):
            logger.info('HTTPS is enabled but cannot find the private key and/or certificate. '
                        'Attempting to generate them')
            private_key = configuration.generate_private_key(api_conf['https']['key'])
            logger.info(f"Generated private key file in WAZUH_PATH/{to_relative_path(api_conf['https']['key'])}")
            configuration.generate_self_signed_certificate(private_key, api_conf['https']['cert'])
            logger.info(f"Generated certificate file in WAZUH_PATH/{to_relative_path(api_conf['https']['cert'])}")
```
Remove final dot for consistency